crates/typst/src/export/pdf/color.rs

```diff
@@ -0,0 +1,382 @@
+pub const HSV: Name<'static> = Name(b"hsv");
+
+/// The name of the HSL color space
+pub const HSL: Name<'static> = Name(b"hsl");
+
+/// The name of the linear RGB color space
+pub const LINEAR_SRGB: Name<'static> = Name(b"linearrgb");
+
+/// The name of the "lightness" component of the OkLab color space
+const OKLAB_L: Name<'static> = Name(b"OkLabL");
+
+/// The name of the "a" component of the OkLab color space
+const OKLAB_A: Name<'static> = Name(b"OkLabA");
+
+/// The name of the "b" component of the OkLab color space
+const OKLAB_B: Name<'static> = Name(b"OkLabB");
+
+/// The name of the "hue" component of the HSV color space
+const HSV_H: Name<'static> = Name(b"HsvH");
+
+/// The name of the "saturation" component of the HSV color space
+const HSV_S: Name<'static> = Name(b"HsvS");
+
+/// The name of the "value" component of the HSV color space
+const HSV_V: Name<'static> = Name(b"HsvV");
+
+/// The name of the "hue" component of the HSL color space
+const HSL_H: Name<'static> = Name(b"HsvH");
+
+/// The name of the "saturation" component of the HSL color space
+const HSL_S: Name<'static> = Name(b"HsvS");
+
+/// The name of the "lightness" component of the HSL color space
+const HSL_L: Name<'static> = Name(b"HsvL");
+
+/// Deflated sRGB ICC profile
+#[comemo::memoize]
+fn srgb_icc() -> Arc<Vec<u8>> {
+    Arc::new(deflate(SRGB_ICC))
+}
+
+/// Deflated gray ICC profile
+#[comemo::memoize]
+fn gray_icc() -> Arc<Vec<u8>> {
+    Arc::new(deflate(GRAY_ICC))
+}
+
+/// Deflated Oklab PostScript function
+#[comemo::memoize]
+fn oklab_function() -> Arc<Vec<u8>> {
+    let code = minify(OKLAB_SOURCE);
+    Arc::new(deflate(code.as_bytes()))
+}
+
+/// Deflated HSV PostScript function
+#[comemo::memoize]
+fn hsv_function() -> Arc<Vec<u8>> {
+    let code = minify(HSV_SOURCE);
+    Arc::new(deflate(code.as_bytes()))
+}
+
+/// Deflated HSL PostScript function
+#[comemo::memoize]
+fn hsl_function() -> Arc<Vec<u8>> {
+    let code = minify(HSL_SOURCE);
+    Arc::new(deflate(code.as_bytes()))
+}
+
+/// This function removes comments, line spaces and carriage returns from a
+/// PostScript program. This is necessary to optimize the size of the PDF file.
+fn minify(source: &str) -> String {
+    let mut buf = String::with_capacity(source.len());
+    let mut in_comment = false;
+    let mut in_line_space = false;
+    for c in source.chars() {
+        match c {
+            '%' => in_comment = true,
+            '\n' => {
+                if !in_line_space {
+                    buf.push(' ');
+                }
+
+                in_comment = false;
+                in_line_space = true;
+            }
+            '\r' => {}
+            _ if in_comment => {}
+            _ if in_line_space => {
+                if !c.is_whitespace() {
+                    in_line_space = false;
+                    buf.push(c);
+                }
+            }
+            _ => {
+                buf.push(c);
+            }
+        }
+    }
+    buf
+}
```

Comment:
May I suggest using `unscanny`?
```suggestion
/// This function removes comments, line spaces and carriage returns from a
/// PostScript program. This is necessary to optimize the size of the PDF file.
fn minify(source: &str) -> String {
    let mut buf = String::with_capacity(source.len());
    let mut s = unscanny::Scanner::new(source);
    while let Some(c) = s.eat() {
        match c {
            '%' => {
                s.eat_until('\n');
            }
            c if c.is_whitespace() => {
                s.eat_whitespace();
                if buf.ends_with(|c: char| !c.is_whitespace()) {
                    buf.push(' ');
                }
            }
            _ => buf.push(c),
        }
    }
    buf
}
```